hail/src/main/scala/is/hail/expr/types/physical/PType.scala

```diff
@@ -295,9 +295,15 @@ abstract class PType extends Serializable with Requiredness {
       case PCanonicalInterval(p, r) =>
         val ti = t.asInstanceOf[TInterval]
         PCanonicalInterval(p.subsetTo(ti.pointType), r)
-      case _ =>
-        assert(virtualType == t)
-        this
+      case _ => {
+        if(t != virtualType && virtualType.isOfType(t)) {
+          PType.canonical(virtualType.deepOptional())
```

Comment:
Probably risky and not what we want, but as a wip test, this does allow the table_union_mixed_requiredeness test to pass

Reply:
we cannot do this, no

Reply:
what error do you hit?